src/map.rs

```diff
@@ -1265,16 +1264,15 @@ where
     /// If `f` returns `false` for a given key/value pair, but the value for that pair is concurrently
     /// modified before the removal takes place, the entry will not be removed.
     /// If you want the removal to happen even in the case of concurrent modification, use [`HashMap::retain_force`].
-    pub fn retain<F>(&mut self, mut f: F)
+    pub fn retain<'g, F>(&mut self, mut f: F, guard: &'g Guard)
```

Comment:
No `'g` necessary. Or add to `&self` and then also include `'g` in the arguments to `FnMut`.